spring-batch-infrastructure/src/main/java/org/springframework/batch/item/ExecutionContext.java

```diff
@@ -273,6 +273,43 @@ public Object get(String key) {
 		return this.map.get(key);
 	}
 
+	/**
+	 * Typesafe getter for the value represented by the provided key, with cast to given class.
+	 *
+	 * @param key The key to get a value for
+	 * @param clazz The class of return type
+	 * @param <V> Type of returned value
+	 * @return The value of given type represented by the given key or {@code null} if the key
+	 * is not present
+	 */
+	@Nullable
+	public <V> V get(String key, Class<V> clazz) {
+		Object value = this.map.get(key);
+		if (value == null) {
+			return null;
+		}
+		return get(key, clazz, null);
+	}
+
+	/**
+	 * Typesafe getter for the value represented by the provided key, with cast to given class.
+	 *
+	 * @param key The key to get a value for
+	 * @param clazz The class of return type
+	 * @param defaultValue Default value in case element is not present
+	 * @param <V> Type of returned value
+	 * @return The value of given type represented by the given key or {@code null} if the key
```

Comment:
I believe this should be "or default value" instead of "or {@code null}". I will fix it on merge.